src/org/loklak/harvester/strategy/BlockingKaizenHarvester.java

```diff
@@ -0,0 +1,61 @@
+package org.loklak.harvester.strategy;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import org.loklak.data.DAO;
+
+public class BlockingKaizenHarvester extends KaizenHarvester {
+
+    /**
+     * This class uses blocking queue to store queries which need to be processed.
+     */
+    public BlockingKaizenHarvester() {
+        super(new KaizenQueries() {
+            private int maxSize = DAO.getConfig("harvester.kaizen.queries_limit", 500);
+            private BlockingQueue<String> queries = new ArrayBlockingQueue<>(maxSize);
+            private int blockingTimeout = DAO.getConfig("harvester.blocking_kaizen.block_time", 120);
+
+            @Override
+            public boolean addQuery(String query) {
+                if (this.queries.contains(query)) {
+                    return false;
+                }
+                try {
+                    /*
+                    * We need to associate a timeout with the addition process as there are chances (though little)
```

Comment:
As I mentioned in the comment, there may be a case when all the harvester threads are trying to put in data at once and there are no consumers. In that case, one of them will get interrupted and drop the query.